sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -569,22 +571,28 @@ private void consumerPollLoop() {
         try {
           if (records.isEmpty()) {
             records = consumer.poll(KAFKA_POLL_TIMEOUT.getMillis());
-          } else if (availableRecordsQueue.offer(
-              records, RECORDS_ENQUEUE_POLL_TIMEOUT.getMillis(), TimeUnit.MILLISECONDS)) {
+            atleastOnePollCompleted.set(true);
+          }
+          if (!records.isEmpty()
+              && availableRecordsQueue.offer(
+                  records, RECORDS_ENQUEUE_POLL_TIMEOUT.getMillis(), TimeUnit.MILLISECONDS)) {
             records = ConsumerRecords.empty();
+            atleastOnePollCompleted.set(true);
           }
-
           commitCheckpointMark();
         } catch (InterruptedException e) {
+          atleastOnePollCompleted.set(true);
```

Comment:
should we instead be throwing or setting consumerPollException in these cases?
I'm a little worried that if we set true once here, that it will then be set false when batch is attempted to be read, and then never gets set to true again and the other thread will block forever.

Instead of the poll completed bit here it seems better to instead note the error state and then have the other thread check for that error state in addition to the atLeastOnePollCompleted and not loop in that case.